build/go.mod

```diff
@@ -1,11 +1,55 @@
 module github.com/mattermost/mattermost-plugin-starter-template/build
 
-go 1.12
+go 1.19
 
 require (
 	github.com/go-git/go-git/v5 v5.4.2
-	github.com/mattermost/mattermost-server/v6 v6.2.1
+	github.com/mattermost/mattermost/server/public v0.0.6
```

Comment:
Question  as a plugin developper: previously to this change, it was easy to know the compatibility between plugin and server version (to know which minimum version to put in the plugin.json file).
With the new package, which seems to have its own independent versioning scheme, how can we find the compatibility table?

Reply:
Thanks for highlighting this, @moussetc! @amyblais / @agnivade, I'm wondering we should plan to tag this submodule at least once a release (ideally at the same commit as the release?), and include that with the release notes.

FWIW, in practice, I think you should be able to use the latest `public` submodule at all times (within the same major version, at least), it's just that not all methods will be supported by the server -- just like today. We do plan to make semantically breaking changes before we hit v1, but only to things we don't expect public consumption of (e.g. our plugin internals that will get refactored back into the core and don't affect compilation or runtime semantics).

Reply:
@agnivade Would you like me to add this to the same ticket (e.g. with https://mattermost.atlassian.net/browse/MM-53676) moving forward? What would an example release note look like?

Reply:
Thanks @lieut-data ! Giving the recommended module version in the server's release notes is perfect :)

Reply:
@amyblais - This is different from bumping dependencies. If we tag a new release for every main release, then there is no need to add a new tag every time we bump the dependencies because anyways it's going to be bumped.

I think this should be part of the task which tags the main release. And lastly, if there are no new commits from the last release, we should not be unnecessarily bumping up the tag of course.

Reply:
Ok, I've opened a ticket to add this to the release tagging process https://mattermost.atlassian.net/browse/CLD-6013.

Reply:
@moussetc I want to point out that the version of `mattermost-server` imported in `go.mod` does not directly coincide with the required `min_server_version` in the plugin's `plugin.json` file. Typically, the available plugin RPC API and hooks needed for the plugin are the actual server requirements. So the `min_server_version` should generally be governed by the use of API and hooks. This way, we can have the `min_server_version` be applicable to more servers running slightly earlier server versions than what is imported.